### PR TITLE
fix(view-engine): load template resources in order specified in view

### PR DESCRIPTION
**Pull request by AshleyGrant**
I'm not sure if this is exactly how we should accomplish this, but this code will force view resources to be loaded sequentially.

This could slow down loading views that have a bunch of resources, but we don't have much choice (other than more intense refactoring) if we want to guarantee resources are loaded sequentially.

Fixes aurelia/bootstrapper#47, aurelia/skeleton-navigation#659

### CODE REVIEW

**Comment by EisenbergEffect** [comment]
@jdanyow Did you have some further thoughts on this? I'm a bit nervous about changing it.

**Comment by AshleyGrant** [comment]
I'm not personally in favor of this change, and I wrote it. That being said, this is definitely something we need to fix, as currently, CSS resources are loaded in to the DOM in the order their requests complete, which is decidedly not ideal.

@jdanyow any chance you have the time on your hands to do this the "right" way?

**Comment by jdanyow** [comment]
These were the thoughts I had:

https://gitter.im/aurelia?at=57f6dddcd6251fd1269aa9e1
https://gitter.im/aurelia?at=57f6e49b68f560d80ce3d4f4

Was thinking we'd track the order a style resource was loaded so we could use the sequence when inserting the style elements in the dom.

I think this would solve the problem but the guidance should probably be "essential style resources like bootstrap" should be bundled and loaded up-front, with a `<link>` tag. Optional style resources that need to be loaded on an as-needed basis could use `<require>`.

In other words, hitting this issue probably means there is some optimization work you could be doing to improve how your styles are loaded.

Does this make sense to you guys?

**Comment by EisenbergEffect** [comment]
Yes. Before we do any work on this though I want to look at the CSS
resource code to see if there is a simpler solution by somehow fixing that.

**Comment by EisenbergEffect** [comment]
Closing this since we're not going to go with this implementation.
